plugins/chat/lib/chat_notifier.rb

```diff
@@ -195,7 +195,7 @@ def group_users_to_notify(users)
     potential_participants, unreachable =
       users.partition do |user|
         guardian = Guardian.new(user)
-        guardian.can_chat? && guardian.can_see_chat_channel?(@chat_channel)
+        guardian.can_chat? && guardian.can_join_chat_channel?(@chat_channel)
       end
```

Comment:
same question about tests

Reply:
Because guardian is tested, I added a test which ensures that guardian is invoked